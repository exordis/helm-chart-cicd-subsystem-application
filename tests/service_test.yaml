# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json

suite: Entities/Services
templates:
  - templates/services.yaml
values:
  - values.metadata.yaml
tests:
  - it: Should generate simple service from minimum data in values 1
    set:
      services:
        myservice:
          ports:
            8081: 
    asserts:
      - matchSnapshot: {}
