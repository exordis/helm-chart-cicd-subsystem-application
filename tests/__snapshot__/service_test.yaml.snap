Should generate simple service from minimum data in values:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations: {}
      labels:
        app.kubernetes.io/component: some-subsystem-some-application
        app.kubernetes.io/instance: ""
        app.kubernetes.io/managed-by: helm
        app.kubernetes.io/name: some-subsystem-some-application
        app.kubernetes.io/part-of: some-subsystem
        app.kubernetes.io/version: 0.0.0
        exordis/application: some-subsystem-some-application
        exordis/application-instance: ""
        exordis/application-type: service
        exordis/environment: test
        exordis/product: Some Product
        exordis/subsystem: some-subsystem
        helm.sh/chart: cicd-subsystem-application-0.1.0
      name: some-subsystem-some-application-myservice
      namespace: some-subsystem-test
    spec:
      ports:
        - name: myservice-8081
          port: 8081
          protocol: TCP
          targetPort: 8081
      selector:
        exordis/application: some-subsystem-some-application
        exordis/application-instance: ""
        exordis/environment: test
        exordis/subsystem: some-subsystem
      sessionAffinity: None
